codecov.yml

```diff
@@ -1 +1,6 @@
 comment: false
+codecov:
+  branch: main
+  require_ci_to_pass: true
+  notify:
+    wait_for_ci: true
```

Comment:
we need to also tell it how many reports it should wait for.

Reply:
Added.... although I'm not sure there's a good way of setting that number

Reply:
this should be the number of builds we send to codecov, and I think we send a lot more than 3. Don't we send on each node of our CI matrix? Isn't this more like 12 or so?

Reply:
I've bumped to 12 (I'm assuming that each individual matrix entry counts as a build to CodeCov?) I'm a bit reluctant to try calculating the limit exactly, since this strikes me as a good way of getting weird behaviour in six months when we've all forgotten about this setting and we decide to drop a Python version.